home/lib/sidekick.sh

```diff
@@ -13,56 +13,84 @@ prompt()
 {
     if [ "${RUN_CWD}" != "$(pwd)" ]; then
         RUN_CWD="$(pwd)"
-        echo -e "\\033[1m[\\033[0m$(pwd)\\033[1m]:\\033[0m"
+        echo -e "\\033[1m[\\033[0m$(pwd)\\033[1m]:\\033[0m" >&2
     fi
 }
 
 run()
 {
-    local COMMAND="$*"
+    local -r COMMAND_DEPRECATED="$*"
+    local -r COMMAND=("$@")
+    local DEPRECATED_MODE=no
+
+    if [[ "${COMMAND[0]}" = *" "* ]]; then
+        DEPRECATED_MODE=yes
+    fi
 
     if [ "$VERBOSE" = "no" ]; then
 
         prompt
-        echo "  > ${COMMAND[*]}"
-        setCommandIndicator $INDICATOR_RUNNING
-
-        if ! bash -c "${COMMAND[@]}" > /tmp/my127ws-stdout.txt 2> /tmp/my127ws-stderr.txt; then
+        if [ "${DEPRECATED_MODE}" = "yes" ]; then
+            echo "  > ${COMMAND_DEPRECATED[*]}" >&2
+            setCommandIndicator "${INDICATOR_RUNNING}"
+            bash -c "${COMMAND_DEPRECATED[@]}" > /tmp/my127ws-stdout.txt 2> /tmp/my127ws-stderr.txt
+        else
+            echo "  >$(printf ' %q' "${COMMAND[@]}")" >&2
+            setCommandIndicator "${INDICATOR_RUNNING}"
+            "${COMMAND[@]}" > /tmp/my127ws-stdout.txt 2> /tmp/my127ws-stderr.txt
+        fi
 
-            setCommandIndicator $INDICATOR_ERROR
+        if [ "$?" -gt 0 ]; then
+            setCommandIndicator "${INDICATOR_ERROR}"
 
             cat /tmp/my127ws-stderr.txt
 
-            echo "----------------------------------"
-            echo "Full Logs :-"
-            echo "  stdout: /tmp/my127ws-stdout.txt"
-            echo "  stderr: /tmp/my127ws-stderr.txt"
+            echo "----------------------------------" >&2
+            echo "Full Logs :-" >&2
+            echo "  stdout: /tmp/my127ws-stdout.txt" >&2
+            echo "  stderr: /tmp/my127ws-stderr.txt" >&2
 
             exit 1
-            
         else
-            setCommandIndicator $INDICATOR_SUCCESS
+            setCommandIndicator "${INDICATOR_SUCCESS}"
         fi
+    elif [ "${DEPRECATED_MODE}" = "yes" ]; then
+        passthru "${COMMAND_DEPRECATED[@]}"
     else
         passthru "${COMMAND[@]}"
     fi
 }
 
 passthru()
 {
-    local COMMAND="$*"
+    local -r COMMAND_DEPRECATED="$*"
+    local -r COMMAND=("$@")
+    local DEPRECATED_MODE=no
+
+    if [[ "${COMMAND[0]}" = *" "* ]]; then
+        DEPRECATED_MODE=yes
+    fi
 
     prompt
 
-    echo -e "\\033[${INDICATOR_PASSTHRU}■\\033[0m > $*"
-    bash -e -c "${COMMAND[@]}"
+    if [ "${DEPRECATED_MODE}" = "yes" ]; then
+        echo -e "\\033[${INDICATOR_PASSTHRU}■\\033[0m > $*" >&2
+        if ! bash -e -c "${COMMAND_DEPRECATED[@]}" >&2; then
```

Comment:
```suggestion
        if ! bash -e -c "${COMMAND_DEPRECATED[@]}"; then
```
We care about the output here

Reply:
stderr cares about it too

Reply:
&2 is stderr not /dev/null

Reply:
oh tho I understand, sorry, you meant the command's output is still meant to be stdout